### PR TITLE
[Shop] Saving new addresses into customer's address book after checkout

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Related tickets | -
| License         | MIT

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
Nice work Jasiek, thanks! :)

**Comment by pjedrzejewski** [comment]
@NeverResponse One note though: have you tried getting rid of the "flush" from the service?

**Comment by liverbool** [comment]
Second note @NeverResponse Can we have someway to improve `AddressComparatorInterface:same` instead of comparison `method by method`?

**Comment by NeverResponse** [comment]
@liverbool This is just a draft, and yeah ten conditions one by one sucks hard, but for now it was the simplest solution available, without sending some heavy (and most likely unnecessary) cavalry.
I'll be working on something simpler, and preferably non-magical.
Although if you already have an idea feel free to fix it.

**Comment by pamil** [comment]
@NeverResponse @liverbool have a look at #6541 🎉

File: features/checkout/having_new_addresses_saved_in_address_book.feature [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        And I am a logged in customer
+        And I have product "Lannister Coat" in the cart
+
+    @ui
+    Scenario: Having the shipping address saved in my address book
+        Given I am at the checkout addressing step
+        When I specify the shipping address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
+        And I complete the addressing step
+        And I proceed with "Free" shipping method and "Cash on Delivery" payment
+        And I confirm my order
```
Comment (michalmarcinkowski) [inline features/checkout/having_new_addresses_saved_in_address_book.feature]: Should be squashed into:
```
Given I am at the checkout addressing step
When I specify the shipping address as "Ankh Morpork", "Frost Alley", "90210", "United States" for "Jon Snow"
And I proceed with "Free" shipping method and "Cash on Delivery" payment
Then I should have 1 address in my address book
```

Here and everywhere else!

File: src/Sylius/Component/Core/Customer/CustomerUniqueAddressAdder.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        $this->customerManager = $customerManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add(AddressInterface $address)
+    {
+        /** @var CustomerInterface $customer */
+        $customer = $this->customerContext->getCustomer();
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Customer/CustomerUniqueAddressAdder.php]: This service shouldn't be statefull, we should be pass the customer as an argument.

**Pull request merged**